src/main/java/net/snowflake/client/jdbc/SnowflakeResultSetSerializableV1.java

```diff
@@ -516,17 +516,20 @@ public static SnowflakeResultSetSerializableV1 create(
     resultSetSerializable.queryId = rootNode.path("data").path("queryId").asText();
 
     JsonNode databaseNode = rootNode.path("data").path("finalDatabaseName");
-    resultSetSerializable.finalDatabaseName = databaseNode.isNull() ? null : databaseNode.asText();
+    resultSetSerializable.finalDatabaseName =
+        databaseNode.isNull() ? sfSession.getDatabase() : databaseNode.asText();
```

Comment:
sfSession is not guaranteed to be non-null. Below this on line 545, we initialize:
`resultSetSerializable.possibleSession = Optional.ofNullable(sfSession);`

Please add a null check for `sfSession` or change to using `possibleSession` to avoid nullptrexceptions

Reply:
Done. Please check,